File: src/dataset/Java/raindrops/raindrops.java
import java.util.HashMap;

class RaindropConverter {
  private static final HashMap<Integer, String> raindrops = new HashMap<>();

  static {
    raindrops.put(3, "Pling");
    raindrops.put(5, "Plang");
    raindrops.put(7, "Plong");
  }

  String convert(int number) {

    return raindrops
        .keySet()
        .stream()
        .filter(key -> number % key == 0)
        .map(raindrops::get)
        .reduce(String::concat)
        .orElse(Integer.toString(number));
  }
}

File: src/dataset/typescript/raindrops/raindrops.ts
type Divider = 3 | 5 | 7;
type Sound = "Pling" | "Plang" | "Plong";

const soundsMap = new Map<Divider, Sound>([
  [3, "Pling"],
  [5, "Plang"],
  [7, "Plong"],
]);

export function convert(input: number): string {
  let converted: string = "";
  soundsMap.forEach((sound, divider) => {
    if (input % divider === 0) {
      converted += sound;
    }
  });

  if (converted.length === 0) {
    return input.toString();
  }

  return converted;
}
